rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -229,6 +230,30 @@ public RMStateFull getRMStateFull(String sessionId) throws NotConnectedException
         return orThrowRpe(RMStateCaching.getRMStateFull());
     }
 
+    @Override
+    public String getModelHosts() throws PermissionRestException {
+        RMStateFull state = orThrowRpe(RMStateCaching.getRMStateFull());
```

Comment:
Are you sure this method does not return REMOVED node events?

Reply:
doubled-checked and REMOVED node or node source events are already filtered from the full state